packages/shared/lib/services/sync/sync.service.ts

```diff
@@ -739,3 +739,10 @@ export async function findPausableDemoSyncs(): Promise<PausableSyncs[]> {
 
     return syncs;
 }
+
+export async function findRecentlyDeletedSync(): Promise<{ id: string }[]> {
+    const q = db.knex.from('_nango_syncs').select('_nango_syncs.id').where(db.knex.raw("_nango_syncs.deleted_at >  NOW() - INTERVAL '6h'"));
```

Comment:
does that mean if the cron doesn't run for more than 6h we will end up missing some syncs to delete?

Reply:
yes as I said in the desc, it's currently not possible to achieve this in an efficient manner. In the best case it will run every 20 minutes so all good but not bullet proof. If you have suggestion I'm taking it.

Reply:
6 hours is not a lot for us to notice if the cron start to fail silently but I don't have a better answer to be honest.

Reply:
I mean I started this change with 24h but there has been about 200 deleted syncs in that range, it's a lot of unnecessary calls every 20minutes. I'm totally okay to put back 24h if you think it's worth it? ☺️

Reply:
your call. you have been looking at this problem way more than me

Reply:
I think I'll delete manually what's left and see how it behaves in prod. Hopefully no further iteration is necessary